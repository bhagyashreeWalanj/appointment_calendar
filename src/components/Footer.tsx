import style from '../styles/Footer.module.scss'

const Footer = () => {
  return (
    <div>
      <footer className={style.footer}>
        <p>
          Crafted with ‚ù§ by
          <a
            className={style.footer_text}
            href="https://github.com/bhagyashreeWalanj"
            target="_blank"
          >
            Bhagyashree Walanj
          </a>
          .
        </p>
        <p>
          <a
            className={style.footer_text}
            href="https://github.com/bhagyashreeWalanj"
            target="_blank"
          >
            Source code
          </a>
          available on GitHub.
        </p>
      </footer>
    </div>
  )
}

export default Footer
