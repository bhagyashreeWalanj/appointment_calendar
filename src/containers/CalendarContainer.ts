import { connect } from 'react-redux'
import { getSlotRequest, scheduleSlotRequest } from '../actions/calendarAction'
import Home from '../components/Home'
import { IAppState } from '../reducers/rootReducer'


// Grab the data from the store and make them available on props
const mapStateToProps = (store: IAppState) => {
  return {
    timeSlots: store.calendar.slots,
    mentor: store.calendar.mentor,
    error: store.calendar.error,
  }
}
const mapDispatchToProps = (dispatch: any) => {
  return {
    fetchAllSlots : ()=> dispatch(getSlotRequest()),
    scheduleCall: (time: string, date: string, status: string)=> dispatch(scheduleSlotRequest(time, date, status))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home as any)
