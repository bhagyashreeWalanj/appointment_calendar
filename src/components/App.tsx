import * as React from 'react'
import CalendarContainer from '../containers/CalendarContainer'
import 'react-calendar/dist/Calendar.css'
import 'react-toastify/dist/ReactToastify.css'
import Footer from './Footer'
import style from '../styles/Home.module.scss'

const App: React.FunctionComponent<{}> = () => {
  return (
    <React.Fragment>
      <h2 className={style.header}>Mentor Calendar</h2>
      <CalendarContainer />
      <Footer />
    </React.Fragment>
  )
}

export default App
