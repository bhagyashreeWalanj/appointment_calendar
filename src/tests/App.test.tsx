import React from 'react'
import App from '../components/App'
import { shallow } from 'enzyme'
import CalendarContainer from '../containers/CalendarContainer'
import Footer from '../components/Footer'

describe('render all components', () => {
  const app = shallow(<App />)

  it('should match the snapshot', () => {
    expect(app).toMatchSnapshot()
  })
  it('App Component contains Home component', () => {
    expect(app.contains(<CalendarContainer />)).toBe(true)
  })
  it('App Component contains Footer component', () => {
    expect(app.contains(<Footer />)).toBe(true)
  })
})
