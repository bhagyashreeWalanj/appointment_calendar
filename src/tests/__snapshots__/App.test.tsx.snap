// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all components should match the snapshot 1`] = `
<Fragment>
  <Connect(Home) />
  <Footer />
</Fragment>
`;
